security/pkg/stsservice/tokenmanager/sts_test.go

```diff
@@ -100,6 +100,39 @@ func TestStsCache(t *testing.T) {
 	}
 }
 
+func TestStsTokenSource(t *testing.T) {
+	stsServer, mockBackend, _ := setUpTestComponents(t, testSetUp{})
+	defer tearDownTest(t, stsServer, mockBackend)
+
+	ts, err := NewTokenSource(mock.FakeTrustDomain, mock.FakeSubjectToken, "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		t.Fatalf("failed to init token source: %v", err)
+	}
+
+	// Override token manager in token source to use mock plugin
+	tokenExchangePlugin, _ := google.CreateTokenManagerPlugin(mock.FakeTrustDomain, mock.FakeProjectNum, mock.FakeGKEClusterURL, false)
+	tokenManager := CreateTokenManager(GoogleTokenExchange,
+		Config{TrustDomain: mock.FakeTrustDomain})
+	tokenManager.(*TokenManager).SetPlugin(tokenExchangePlugin)
+	ts.setTokenManager(tokenManager)
+
+	// Get and verify access token
+	got, err := ts.Token()
+	if err != nil {
+		t.Fatalf("failed to get access token: %v", err)
+	}
+	if got.AccessToken != mock.FakeAccessToken {
+		t.Errorf("access token want %v got %v", mock.FakeAccessToken, got.AccessToken)
+	}
+	if got.TokenType != "Bearer" {
+		t.Errorf("access token type want %v got %v", "Bearer", got.TokenType)
+	}
+	expireIn := time.Until(got.Expiry)
+	if expireIn < 3500*time.Second {
```

Comment:
@JimmyCYJ are we already having test like this? this is using a unexported implementation details in the mock token

Reply:
Exposed expiry duration in the mock server package.